index c58df340..21862a63 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -58,8 +58,8 @@ bazel_dep(name = "score_rules_imagefs", version = "0.0.3", dev_dependency = True

 bazel_dep(name = "score_docs_as_code", version = "4.6.1")

-bazel_dep(name = "score_platform", version = "0.6.2", dev_dependency = True)
-bazel_dep(name = "score_process", version = "2.0.2", dev_dependency = True)
+bazel_dep(name = "score_platform", version = "0.6.2")
+bazel_dep(name = "score_process", version = "2.0.2")

 bazel_dep(name = "score_python_basics", version = "0.3.4")
 bazel_dep(name = "score_tooling", version = "1.2.0")
 