diff --git a/BUILD b/BUILD
--- a/BUILD
+++ b/BUILD
@@ -11,8 +11,7 @@
 # SPDX-License-Identifier: Apache-2.0
 # *******************************************************************************
 load("@score_docs_as_code//:docs.bzl", "docs")
-load("@score_tooling//:defs.bzl", "copyright_checker", "setup_starpls")
-load("@score_tooling//third_party/format:macros.bzl", "use_format_targets")
+load("@score_tooling//:defs.bzl", "copyright_checker", "setup_starpls", "use_format_targets")
 
 setup_starpls(
     name = "starpls_server",
